lib/go/edgecontext/edgecontext_test.go

```diff
@@ -243,85 +243,61 @@ func TestNew(t *testing.T) {
 }
 
 func TestLocale(t *testing.T) {
-	t.Run(
-		"valid-language",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("es")
-
-			if err != nil {
-				t.Fatal(err)
-			}
+	for _, c := range []struct {
+		label  string
+		locale string
+		valid  bool
+	}{
+		{
+			label:  "valid-language",
+			locale: "es",
+			valid:  true,
 		},
-	)
-
-	t.Run(
-		"valid-language-valid-region",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("es_MX")
-
-			if err != nil {
-				t.Fatal(err)
-			}
+		{
+			label:  "valid-language-valid-region",
+			locale: "es_MX",
+			valid:  true,
 		},
-	)
-
-	t.Run(
-		"valid-language-invalid-region",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("es_MEX")
-
-			if err == nil {
-				t.Errorf("Expected InvalidLocaleCodeError")
-			}
+		{
+			label:  "valid-language-invalid-region",
+			locale: "es_MEX",
+			valid:  false,
 		},
-	)
-
-	t.Run(
-		"invalid-language-valid-region",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("esp_MX")
-
-			if err == nil {
-				t.Errorf("Expected InvalidLocaleCodeError")
-			}
+		{
+			label:  "invalid-language-valid-region",
+			locale: "esp_MX",
+			valid:  false,
 		},
-	)
-
-	t.Run(
-		"invalid-language-invalid-region",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("esp_MEX")
-
-			if err == nil {
-				t.Errorf("Expected InvalidLocaleCodeError")
-			}
+		{
+			label:  "invalid-language-invalid-region",
+			locale: "esp_MEX",
+			valid:  false,
 		},
-	)
-
-	t.Run(
-		"invalid-separator",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("es-MX")
-
-			if err == nil {
-				t.Errorf("Expected InvalidLocaleCodeError")
-			}
+		{
+			label:  "invalid-separator",
+			locale: "es-MX",
+			valid:  false,
 		},
-	)
-
-	t.Run(
-		"invalid-capitalization",
-		func(t *testing.T) {
-			err := EdgeContextWithLocaleCode("ES-MX")
-
-			if err == nil {
-				t.Errorf("Expected InvalidLocaleCodeError")
-			}
+		{
+			label:  "invalid-capitalization",
+			locale: "ES_MX",
+			valid:  false,
 		},
-	)
+	} {
+		t.Run(c.label, func(t *testing.T) {
+			err := edgeContextWithLocaleCode(c.locale)
+			if (err == nil) != c.valid {
+				if !c.valid {
+					t.Errorf("Expected InvalidLocaleCodeError")
```

Comment:
nit: there's no format string so you can use `t.Error` instead of `t.Errorf`.

also, if the expected error is always `ErrInvalidLocaleCode`, then we should also check for that (currently we just checked that err is not nil).